vnext/Microsoft.ReactNative.Cxx/CoreApp.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <roapi.h>
+
+#ifdef __cplusplus
+#define NORETURN [[noreturn]]
+
+struct RNCoreApp {
```

Comment:
>struct [](http://example.com/codeflow?start=0&length=6)

This struct is a part of the public API. It must be C-only POD struct. We must not use C++ classes in the public ABI-safe API. Also, for a struct to be ABI-safe we need to add some kind of version. It is either a version field or the struct size as the first field.

It is also a bit unclear which fields are provided by the framework and which must be provided by the app.

Should we rather have two different structs?